### PR TITLE
Add lal

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR adds the LAL package, providing core gravitational wave analysis routines.

cc @skymoo

### CODE REVIEW

File: recipes/lal/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - swig>=3.0.7
+  host:
+    - gsl
+    - zlib
+    - fftw
+    - hdf5
+    - python
+    - numpy
+    - six
+    - lscsoft-glue
```
Comment (mariusvniekerk) [inline recipes/lal/meta.yaml]: In order to silence some of the conda build patch warnings we may want to add blas in here.

File: recipes/lal/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - pkg-config
+    - make
+    - {{ compiler('c') }}
+    - python {{ python }}
```
Comment (mariusvniekerk) [inline recipes/lal/meta.yaml]: Do we need this given that we have python in the host section already?

File: recipes/lal/meta.yaml (hunk 1 above)
Comment (skymoo) [reply recipes/lal/meta.yaml]: `blas` is a required dependency but we currently get this [through](https://git.ligo.org/lscsoft/lalsuite/blob/c302e263bb5ad38df8209f3096f9db9fa3f89a0c/lal/configure.ac#L212) `gsl`. We are currently investigating how to support multiple different `blas` libraries.

Comment (duncanmmacleod) [reply recipes/lal/meta.yaml]: @mariusvniekerk, should we add an explicit dependency on `openblas` or `blas` in `lal/meta.yaml`, or is @skymoo's explanation enough?

**Comment by duncanmmacleod** [comment]
@mariusvniekerk, is this ready to be merged now? Apologies if its not the done thing to nag about PRs (I know there are a lot of PRs for this project, so if everyody nagged, that would be ... annoying).

**Pull request merged**